### ISSUE TITLE
can provide win64 build (mingw)

### ISSUE THREAD

**Issue by djusHa** [Flameeyes/unpaper#52]
Any interests?

Build from current master Branch a few Days ago with mingw64 and generated Docs (html format).

**Comment by zone-b** [comment]
It would be great if you could provide instructions how to build unpaper on windows!

**Comment by Flameeyes** [comment]
Yeah I think I'd take documentation over binaries -- I don't want to get into the business of verifying binaries, so if you want to provide them you're welcome to, but wouldn't want to "sign off" on them.

Instructions OTOH would be appreciated.

**Comment by djusHa** [comment]
No Problem, will provide build Instruction later on.

**Comment by zone-b** [comment]
Yeah, documentation will be GREAT!

**Comment by zone-b** [comment]
@djusHa: maybe you can give me some short hints and I will conduct it?

**Comment by djusHa** [comment]
Hi @everyone.

Just forgot about this, sorry ...

By the Way, best tool for windows 'cmd' replacement, is IMHO [ConEmu](https://conemu.github.io/) or a more
'sexier' version [Cmder](http://cmder.net/).

You can easily setup MSYS2 to run with.

Just add `-conemu` to your MSYS2 MinGW Shortcut, finally it should looks like
`C:\msys64\msys2_shell.cmd -mingw64 -conemu`.

## First you need a fresh install File of MSYS2:
[MSYS2](http://www.msys2.org/)
install as usual.

Open a mingw 32 or 64 Console (there are three Consoles on x64 Build) and
update MSYS2 / Pacman:

`pacman -Sy pacman`

`pacman -Syu`

`pacman -Su`

please follow the Instructions on Console Window after each step!

## Install Buildtools (yes, its one line):
In MinGW 32 or 64 Console, type

`pacman -S base-devel mingw-w64-x86_64-toolchain diffutils pkg-config git mingw-w64-x86_64-yasm mingw-w64-x86_64-gcc mingw-w64-x86_64-SDL`

Install ALL Components!

If you want a x86 Build, please install only Packages with 'i686' in name, means mingw-w64-**i686**-toolchain and not base-devel mingw-w64-**x86_64**-toolchain and so on!


## Get and build libav

`pacman -S make pkgconf diffutils`

`curl -O https://libav.org/releases/libav-11.8.tar.xz`

`tar -xf libav-11.8.tar.xz` --> Ignore symlink error, if any

`cd libav-11.8`

`./configure --target-os=mingw32 --prefix=${MINGW_PREFIX}` --> takes a while...

`make -r` --> takes about 5 Minutes on my Core i7-6700 ;-)

`make install`


## Get Unpaper and build it

`cd ~`

`git clone https://github.com/Flameeyes/unpaper.git ./unpaper`

`cd unpaper`

`autoreconf -fi`

`mkdir exe`

`./configure --prefix=${PWD}/exe`

`make -r`

`make install`

You're done :-)

unpaper.exe should be in `C:\msys64\home\**yourusername**\unpaper\exe\bin`.

If you plan to run unpaper 'native', the following Files **MUST** be located in the same directory!

`c:\msys64\mingw64\bin\LIBBZ2-1.DLL`
`c:\msys64\mingw64\bin\LIBWINPTHREAD-1.DLL`
`c:\msys64\mingw64\bin\ZLIB1.DLL`

So long...

Sorry for my poor english, please feel free to correct my mistakes.

**Comment by zone-b** [comment]
@djusHa: it works like a charm! Great contribution, thank you! 🥇

**Comment by djusHa** [comment]
@zone-b: you welcome, thanks for feedback :-)

**Comment by Artgit** [comment]
@djusHa thanks for your instructions!

Unfortunately, I'm failed on the following step:

`./configure --target-os=mingw32 --prefix=${MINGW_PREFIX}`

with the following error:

```
gcc is unable to create an executable file.
If gcc is a cross-compiler, use the --enable-cross-compile option.
Only do this if you know what cross compiling means.
C compiler test failed.

If you think configure made a mistake, make sure you are using the latest
version from Git.  If the latest version fails, report the problem to the
libav-tools@libav.org mailing list or IRC #libav on irc.freenode.net.
Include the log file "config.log" produced by configure as this will help
solving the problem.
```
What am I doing wrong and how to fix it ?

**Comment by djusHa** [comment]
Hi, no Plan what is goin wrong. Seems to be a Problem with libav. Did you download the required Version?
`curl -O https://libav.org/releases/libav-11.8.tar.xz`

**Comment by djusHa** [comment]
There was an error in the Guide above, run
`pacman -S make pkgconfig diffutils`
instead of
`pacman -S make pkgconf diffutils`

**Comment by suzannealice** [comment]
Thanks for the great instruction! Unfortunately I get a fatal error in the ./configure --target-os=mingw32 --prefix=${MINGW_PREFIX} step: 
CC      cmdutils.o
cmdutils.c:49:10: fatal error: version.h: No such file or directory
 #include "version.h"
          ^~~~~~~~~~~
compilation terminated.
make: *** [Makefile:44: cmdutils.o] Error 1

It seems like I am missing a certain file, what am I doing wrong? Thanks in advance!

**Comment by djusHa** [comment]
@suzannealice 
Hi, i'm not a c expert, but possibly there is an error in cmdutils.c
Try to edit cmdutils.c and change line 49 from
`#include "cmdutils.h"`
to
`#include "libavutil/version.h"`

**Comment by djusHa** [comment]
@suzannealice 
Any luck?

**Comment by suzannealice** [comment]
@djusHa Kind of, I am now some steps further but now run into the error: configure: error: C compiler cannot create executables

Do you have any experience with that?

**Comment by Geethebluesky** [comment]
@djusHa 
I tried building Unpaper with your instructions, but ended up at a wall, see my post on Stackoverflow here:

https://stackoverflow.com/questions/53908046/deciphering-setup-issues-when-attempting-to-build-things-in-msys2-with-libav-11

I suppose my lack of skill here is at fault.

Would you still be able to provide a Win64 build of Unpaper and share it somewhere please?

**Comment by djusHa** [comment]
Hi,
no problem, here we go ( version 6.2 ):
http://www.mediafire.com/file/z7akmon171wdqvb/unpaper.7z/file

**Comment by rodrigost23** [comment]
@djusHa I followed your build instructions for Windows and created a [workflow](https://github.com/rodrigost23/unpaper/blob/master/.github/workflows/windows.yml) using GitHub Actions

It should generate new builds automatically from tags (6.1 available): https://github.com/rodrigost23/unpaper/releases

And you can download a build based on the latest commit here: https://github.com/rodrigost23/unpaper/actions/runs/124159600

It's only 64-bit for now, but I may update it later

**Comment by djusHa** [comment]
@rodrigost23: Really cool man, this is very helpful!
